src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -315,5 +388,43 @@ private static LogFormat ComputeLogFormat(IList<MediaTypeHeaderValue> acceptedHe
             }
             return LogFormat.None;
         }
+
+        private static Func<Stream, CancellationToken, Task> ConvertFastSerializeAction(Func<CancellationToken, Task<IFastSerializable>> action)
+        {
+            return async (stream, token) =>
+            {
+                IFastSerializable fastSerializable = await action(token);
+
+                // FastSerialization requests the length of the stream before serializing to the stream.
+                // If the stream is a response stream, requesting the length or setting the position is
+                // not supported. Create an intermediate buffer if testing the stream fails.
+                bool useIntermediateStream = false;
+                try
+                {
+                    _ = stream.Length;
```

Comment:
Is there a case where this succeeds? If not, we should probably just always use the intermediate stream.

Reply:
Works for any stream that has a length, such a file streams and memory streams.